src/loginComponents/checkRoles.js

```diff
@@ -11,10 +11,7 @@ export class AuthorizedElement extends React.Component {
   }
 
   render() {
-    if (
-      this.props.fatherState.keycloak.hasRealmRole(this.props.roleNeeded) &&
-      this.props.fatherState.groups.indexOf(this.props.groupNeeded) > -1
-    ) {
+    if (this.props.iSuperAdmin) {
```

Comment:
I think the best way to keep this code generic enough is an something like:

```
<AuthorizedElement probs={props} tenantNeeded={"Tenant1"} roleNeeded={"tenant-admin"} groupNeeded={"admins"} superAdminCheck={"false"}></AuthorizedElement>
```

Then inside you can implement a logic that check what is in token based on what is configured in `tenantNeeded`, `superAdminCheck`, `roleNeeded`, `groupNeeded`.

so if tenantNeeded is null we don't check the tenant, if it is we require that the token includes the tenant, ect.